src/services/MetaService.ts

```diff
@@ -85,7 +97,32 @@ export class MetaService {
     return this.extract(requested);
   }
 
+  /**
+   * Make http request to get track data. Response data will be parsed, then the Promise will be resolved.
+   */
+  async getTracks(): Promise<BullhornTrack[]> {
+    await this.initialized; // This is ensuring that our initialization is complete
```

Comment:
Also not a big deal one way or another. But I don't think you really need the comment here. I think it's pretty clear with the naming of this.initialized.

Reply:
I wanted to leave that in there so future developers can see that they need to make sure this is resolved before they do other things. If you think we don't need it, I can take it out.

Reply:
Taking it out

Reply:
Ah that makes sense. No it's totally fine. I will resolve this comment.